llvm/lib/Transforms/InstCombine/InstCombineAddSub.cpp

```diff
@@ -2631,6 +2631,18 @@ Instruction *InstCombinerImpl::visitSub(BinaryOperator &I) {
                                                /* IsNUW */ false))
       return replaceInstUsesWith(I, Res);
 
+  if (match(Op0, m_ZExt(m_PtrToInt(m_Value(LHSOp)))) &&
+      match(Op1, m_PtrToInt(m_Value(RHSOp))) && isa<GlobalValue>(RHSOp)) {
```

Comment:
I don't see a reason to require a global here.

Reply:
We should really be matching `m_ZExt(m_PtrToInt())` on the RHS as well. And adjust InstCombine to produce that. The fact that the ptrtoint is not converted into zext + ptrtoint is unintended fallout from constexpr removal.

Reply:
Hm, it's probably fairly tricky to do that conversion, so using m_ZExtOrSelf here would be fine for now.